src/lib/stitching/exchange/transformers/replaceCommerceDateTimeType.ts

```diff
@@ -67,8 +69,11 @@ export class ReplaceCommerceDateTimeType implements Transform {
     const newFields: GraphQLFieldConfigMap<any, any> = {}
     const resolveType = createResolveType((_name, type) => {
       if (
-        type.name === "CommerceDateTime" ||
-        (type.ofType && type.ofType.name === "CommerceDateTime")
+        isNamedType(type) &&
+        (type.name === "CommerceDateTime" ||
+          (isWrappingType(type) &&
+            type.ofType &&
+            type.ofType.name === "CommerceDateTime"))
```

Comment:
This is just adding some TS type guard functions, which besides checking the actual expected type also narrow the TS type definition down to the expected type (which is one that has `name` or `ofType`).